### PR TITLE
[DS-3602] Upgrade id fields in statistics records to UUIDs

### PR DESCRIPTION
**Pull request by terrywbrady**
https://jira.duraspace.org/browse/DS-3602

Please see the Jira ticket for a discussion of possible approaches to this problem.

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/util/StatisticsMigrator.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+import org.dspace.content.service.CollectionService;
+import org.dspace.content.service.CommunityService;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Constants;
+import org.dspace.core.Context;
+
+import au.com.bytecode.opencsv.CSVReader;
+import au.com.bytecode.opencsv.CSVWriter;
+
+public class StatisticsMigrator {
```
Comment (tdonohue) [inline dspace-api/src/main/java/org/dspace/util/StatisticsMigrator.java]: If this Migrator is very specific to DSpace 5 to 6, it likely needs to be named as such (unless you feel this has a more generic use case and may be a migrator for future versions as well).  If this is 6 specific, it likely should be renamed `Statistics6Migrator` (or similar) and the cli flag changed to `upgrade6` or similar.

**Comment by tdonohue** [comment]
This looks good overall. Added one inline comment. 

One sidenote here though...the current downside of this implementation is that we'd need clear documentation to warn folks to run this *manually* as it won't run automatically. 

Therefore, another option to consider here is to turn this into a Flyway Java migration that runs automatically when upgrading to DSpace 6. Though the downside with that option is that will kick off automatically when DSpace starts...and lock up the site until it finishes.  

So, if the goal here is to run this "behind the scenes" (while the site is up), then your implementation may be more ideal.  But, if we'd rather this be automated, this probably should be turned into a Flyway Java migration.

**Comment by terrywbrady** [comment]
See Jira for some on-going discussion of an alternative approach.  I am going to attempt an alternative approach that will incrementally update SOLR stats records.

**Comment by terrywbrady** [comment]
Closing in favor of https://github.com/DSpace/DSpace/pull/1766

**Pull request closed without merging**